### PR TITLE
Add global setting to disable keyboad shortcuts

### LINKED ISSUE #34079: Accessibility settings are cut off with 3 line "buttons"
Bildschirmfoto vom 2022-09-14 15-42-32

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Add global setting in accessibility settings
- [x] Respect for ctrl+f search
- [x] Respect for closing menus with Escape
- [ ] Raise follow up issues for other shortcuts, but allow this PR to get in so apps can adapt in time
- [ ] Add to https://github.com/nextcloud/server/issues/32117 once merged

This is a pre-requirement for https://github.com/nextcloud/spreed/issues/7563

Fix #34079

### CODE REVIEW

File: apps/theming/src/UserThemes.vue [hunk 1]
```diff
@@ -122,6 +141,31 @@ export default {
 			this.selectItem(enabled, id)
 		},
 
+		async changeShortcutsDisabled(newState) {
+			console.error('changeShortcutsDisabled', newState)
```
Comment (CarlSchwan) [inline apps/theming/src/UserThemes.vue:145]:
```suggestion
```

leftover?

Comment (nickvergessen) [reply apps/theming/src/UserThemes.vue:145]: yes. Let's see if there is more feedback to avoid recompiling and CI time before pushing it away

**Review by CarlSchwan — APPROVED** [review]
Fine by me, make sure if you write doc about this that `loadState('theming', 'shortcutsDisabled', false),` needs to have a default value otherwise this will break when the theming app is disabled

**Review by PVince81 — APPROVED** [review]
:+1: don't forget to remove the stray console.error

**Comment by szaimen** [comment]
Not sure if we should merge this before https://github.com/nextcloud/server/pull/33733

File: apps/theming/src/UserThemes.vue (hunk 1 above)
Comment (nickvergessen) [reply apps/theming/src/UserThemes.vue]: Fixed

**Comment by nickvergessen** [comment]
Rebased after #33733 

And now with a global OCP method as discussed in https://github.com/nextcloud/contacts/issues/2827#issuecomment-1247646650 and chat

File: apps/theming/lib/Listener/BeforeTemplateRenderedListener.php [hunk 2]
```diff
@@ -61,9 +63,20 @@
...
 
+		/** @var BeforeTemplateRenderedEvent $event */
+		if ($event->getResponse()->getRenderAs() === TemplateResponse::RENDER_AS_USER) {
+			$this->initialState->provideLazyInitialState('shortcutsDisabled', function () {
+				if ($this->userSession->getUser()) {
+					$uid = $this->userSession->getUser()->getUID();
+					return $this->config->getUserValue($uid, Application::APP_ID, 'shortcuts_disabled', 'no') === 'yes';
+				}
+				return false;
+			});
```
[the remark this answers is not in the document]
Comment (nickvergessen) [reply apps/theming/lib/Listener/BeforeTemplateRenderedListener.php:77]: I think it's an oversight that https://github.com/nextcloud/server/blame/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/lib/public/AppFramework/Services/IInitialState.php#L60 is missing `Closure():bool`. The non-lazy method clearly states `bool` as well?

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**